core/src/main/scala/org/bykn/bosatsu/Normalization.scala

```diff
@@ -398,6 +443,16 @@ case class NormalizePackageMap(pm: PackageMap.Inferred) {
     PackageMap(normAll.run(Map()).value._2.toMap)
   }
 
+  def hashKey[T](fn: NormalExpression => T): PackageMap.Typed[(Declaration, ExpressionKeyTag[T])] = {
+    val lst = normalizePackageMap.toMap.toList
+      .map { case (name, pack) => (name, pack.copy(program = pack.program.copy(
+        lets = pack.program.lets.map { case (name, recursive, expr) => (name, recursive, expr.traverse[Id, (Declaration, ExpressionKeyTag[T])] {
```

Comment:
this is a bit dense. Can we expand it into a few lines with a few `val =`? I'd rather use a bit more vertical than horizontal space here.